Add basic random color palette generator

michaelfiber opened the pull request:
This does a bit of fiddling with HSV values as a way to make quick and dirty color palettes that can randomly be monotone, complementary or split complementary.  It isn't guaranteed to make a nice palette or even a legible one.  But if you click and you can't read any more, just click again without moving the mouse!

**Note**: I modified the Makefile a bit so that I could just do `RAYLIB_PATH=/home/myprofile/dev/raylib make all` to build. If that breaks anything else I can undo it.

raysan5 commented:
@michaelfiber Thanks for the addition! I reviewed the files and you are adding the binary, it shouldn't be there.

Also, I follow the same [raylib coding conventions](https://github.com/raysan5/raylib/blob/master/CONVENTIONS.md) for `rGuiStyler`, could you review the formatting?

raysan5 commented on src/gui_main_toolbar.h:
```diff
@@ -189,9 +189,9 @@ void GuiMainToolbar(GuiMainToolbarState *state)
     state->btnSaveFilePressed = GuiButton((Rectangle){ state->anchorFile.x + 12 + 48 + 8, state->anchorFile.y + 8, 24, 24 }, "#6#");
     state->btnExportFilePressed = GuiButton((Rectangle){ state->anchorFile.x + 12 + 72 + 12, state->anchorFile.y + 8, 24, 24 }, "#7#");
-    GuiDisable();
+    //GuiDisable();
```
It can be just deleted.

michaelfiber commented:
I removed the binary, removed the commented out guienable/disable stuff, and went over trying to match the style guide.  Let me know if there's anything else.

raysan5 commented:
@michaelfiber Thank you very much for the addition! :D
